agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2535,9 +2535,37 @@ bool backhaul_manager::handle_slave_ap_metrics_response(ieee1905_1::CmduMessageR
             metric.estimated_service_info_field.push_back(info[i]);
         }
     }
+    std::vector<sStaTrafficStats> traffic_stats_response;
+
+    for (auto sta_traffic : cmdu_rx.getClassList<wfa_map::tlvAssociatedStaTrafficStats>()) {
+        if (!sta_traffic) {
+            LOG(ERROR) << "Failed to get class list for tlvAssociatedStaTrafficStats";
+            continue;
+        }
+
+        traffic_stats_response.push_back(
+            {sta_traffic->sta_mac(), sta_traffic->byte_sent(), sta_traffic->byte_recived(),
+             sta_traffic->packets_sent(), sta_traffic->packets_recived(),
+             sta_traffic->tx_packets_error(), sta_traffic->rx_packets_error(),
+             sta_traffic->retransmission_count()});
+    }
+
+    std::vector<sStaApLinkMetrics> link_metrics_response;
+    for (auto sta_link_metric : cmdu_rx.getClassList<wfa_map::tlvAssociatedStaLinkMetrics>()) {
+        if (!sta_link_metric) {
+            LOG(ERROR) << "Failed getClassList<wfa_map::tlvAssociatedStaLinkMetrics>";
+            continue;
+        }
+        if (sta_link_metric->bssid_info_list_length() != 1) {
+            LOG(ERROR) << "sta_link_metric->bssid_info_list_length() should be equal to 1";
+            continue;
+        }
+        auto response_list = sta_link_metric->bssid_info_list(0);
+        link_metrics_response.push_back({sta_link_metric->sta_mac(), std::get<1>(response_list)});
+    }
 
     // Fill a response vector
-    m_ap_metric_response.push_back({metric});
+    m_ap_metric_response.push_back({metric, traffic_stats_response, link_metrics_response});
```

Comment:
Maybe it's simpler and easier to maintain if you create and store the TLVs instead of intermediate structures

Reply:
Can we create tlv without cmdu_rx/tx objects?

Reply:
cmdu_tx is defined in base class `beerocks::btl::socket_thread`, but you could define your own message here and fill it (instead of using those intermediate structures).

Reply:
It will be done as part of https://github.com/prplfoundation/prplMesh/issues/1328